### PR TITLE
Pocketxtra 21 component expandable

### PR DESCRIPTION
**Pull request by tabassum2507**
@rohan-arya-w Please review the expandable components

### CODE REVIEW

File: src/components/expandable/ExpandableWithInputs.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  height = 20,
+  bgColor = "",
+  borderRadius = 3,
+  borderWidth = 1,
+  borderColor = "",
+  IconColor = '',
+  textColor = "",
+  textSize = 2,
+  IconSize = 7,
+  fontFamily = "Nunito"
```
Comment (rohan-arya-w) [inline src/components/expandable/ExpandableWithInputs.tsx]: Remove this

File: src/components/expandable/ExpandableWithInputs.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+          borderColor: borderColor, 
+          borderRadius: responsiveWidth(borderRadius),   
+          borderWidth: responsiveWidth(borderWidth),
+          width: responsiveWidth(width),
+          height: responsiveWidth(height),
+          alignSelf : 'center',
+          backgroundColor: bgColor,               
+      }}
+      right={props => (
+        <View style={{ flexDirection: 'row', alignItems: 'center', justifyContent: 'center' }}>
```
Comment (rohan-arya-w) [inline src/components/expandable/ExpandableWithInputs.tsx]: Remove inline styling

File: src/components/expandable/ExpandableWithSelection.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    responsiveHeight,
+    responsiveWidth,
+    responsiveFontSize
+} from "react-native-responsive-dimensions";
+import { expandableWithSelectionInterface } from '../../interface/expandable/expandableWithSelectionInterface';
+// import imagePath from '../../../assets/favicon.png'; 
+
+export const ExpandableWithSelection: React.FC<expandableWithSelectionInterface> = ({
+    Icon = 'checkmark-circle',
+    IconInfo = "information-circle-outline",
```
Comment (rohan-arya-w) [inline src/components/expandable/ExpandableWithSelection.tsx]: Remove the hardcoded data

File: src/components/expandable/ExpandableWithSelection.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+                            paddingBottom: 25,
+                            backgroundColor: backgroundColour,
+                            
+                        }}
+
+                    > 
+                    <Image
+                      source={{ uri: imageSource.uri }}
+                      style={{
+                        height: responsiveWidth(40),
```
Comment (rohan-arya-w) [inline src/components/expandable/ExpandableWithSelection.tsx]: remove in line styles

File: src/components/expandable/ExpandableWithInputs.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  const [expanded, setExpanded] = React.useState(true);
+  const [inputValue, setInputValue] = React.useState('');
+
+  const handlePress = () => setExpanded(!expanded); 
+return (
+  <View style={{  width: responsiveWidth(width), height: responsiveWidth(height)}}>
+     <List.Section >
+     <List.Accordion
+      title= {title}
+      titleStyle={{color : textColor, fontSize : responsiveFontSize(textSize), alignItems: 'center', justifyContent: 'center', fontFamily: fontFamily}}
```
Comment (rohan-arya-w) [inline src/components/expandable/ExpandableWithInputs.tsx]: remove font family

File: src/components/expandable/ExpandableWithSelection.tsx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import Pen from 'react-native-vector-icons/Ionicons';
+import { List } from 'react-native-paper';
+import {
+    responsiveHeight,
+    responsiveWidth,
+    responsiveFontSize
+} from "react-native-responsive-dimensions";
+import { expandableWithSelectionInterface } from '../../interface/expandable/expandableWithSelectionInterface';
+// import imagePath from '../../../assets/favicon.png';
```
Comment (rohan-arya-w) [inline src/components/expandable/ExpandableWithSelection.tsx]: Remove comments

File: src/components/expandable/ExpandableWithSelection.tsx [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+                            {...props}
+                            name={Icon}
+                            size={responsiveWidth(IconSize)}
+                            color={penColor}
+                            style={{ alignItems: 'center', justifyContent: 'center', marginTop: responsiveWidth(-3) }}
+
+
+                        />
+                       
+
```
Comment (rohan-arya-w) [inline src/components/expandable/ExpandableWithSelection.tsx]: Format this file

File: src/components/expandable/ExpandableWithSelection.tsx [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+                        height: responsiveWidth(40),
+                        width: responsiveWidth(40),
+                        alignSelf: 'center',
+                        marginBottom: responsiveWidth(10),
+                        marginTop: responsiveWidth(10),
+                        marginRight: responsiveWidth(10),
+                      }}
+                    />
+                  
+                        <Text style={{ color: exTextColor,  alignItems: 'flex-end', fontSize: responsiveFontSize(exTextSize), marginBottom: responsiveWidth(3), fontFamily: fontFamily }}>
```
Comment (rohan-arya-w) [inline src/components/expandable/ExpandableWithSelection.tsx]: remove in line styles

File: src/components/expandable/ExpandableWithSelection.tsx [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+                            {userRole}
+                        </Text>
+                        <Text style={{ color: exTextColor,  marginRight: responsiveWidth(10), fontFamily: fontFamily }}>
```
Comment (rohan-arya-w) [inline src/components/expandable/ExpandableWithSelection.tsx]: write proper name

File: src/components/expandable/ExpandableWithSelection.tsx [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+                            {expanablePara}
```
Comment (rohan-arya-w) [inline src/components/expandable/ExpandableWithSelection.tsx]: Correct this spelling

File: src/interface/expandable/expandableWithSelectionInterface.ts [hunk 11]
```diff
@@ -0,0 +1,27 @@
+export interface expandableWithSelectionInterface{
+    Icon?: string,
+    IconInfo?: string,
```
Comment (rohan-arya-w) [inline src/interface/expandable/expandableWithSelectionInterface.ts]: Follow camel case

File: src/interface/expandable/expandableWithSelectionInterface.ts [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    title?: string,
+    expanablePara?: string,
```
Comment (rohan-arya-w) [inline src/interface/expandable/expandableWithSelectionInterface.ts]: check spelling

**Review by rohan-arya-w — CHANGES_REQUESTED** [review]
Remove all in line styling

File: src/components/expandable/ExpandableWithInputs.tsx [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+import { List } from 'react-native-paper';
+import {
+    responsiveHeight,
+    responsiveWidth,
+    responsiveFontSize
+} from "react-native-responsive-dimensions";
+import { expandableInputInterface } from '../../interface/expandable/expandableInputInterface';
+
+export const ExpandableWithInputs: React.FC<expandableInputInterface> = ({
+  Icon = '',
```
Comment (rohan-arya-w) [inline src/components/expandable/ExpandableWithInputs.tsx]: Follow double quotes

File: src/components/expandable/ExpandableWithInputs.tsx [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+  title= "",
+  height = 20,
+  bgColor = "",
+  borderRadius = 3,
+  borderWidth = 1,
+  borderColor = "",
+  IconColor = '',
+  textColor = "",
+  textSize = 2,
+  IconSize = 7,
```
Comment (rohan-arya-w) [inline src/components/expandable/ExpandableWithInputs.tsx]: Follow camel case

**Review by rohan-arya-w — CHANGES_REQUESTED** [review]
Check for inline styles and hardcoded values

File: src/components/expandable/ExpandableWithInputs.tsx (hunk 1 above)
Comment (tabassum2507) [reply src/components/expandable/ExpandableWithInputs.tsx]: removed

File: src/components/expandable/ExpandableWithSelection.tsx (hunk 3 above)
Comment (tabassum2507) [reply src/components/expandable/ExpandableWithSelection.tsx]: removed

File: src/components/expandable/ExpandableWithSelection.tsx (hunk 9 above)
Comment (tabassum2507) [reply src/components/expandable/ExpandableWithSelection.tsx]: done

File: src/components/expandable/ExpandableWithSelection.tsx [hunk 15]
```diff
@@ -0,0 +1,173 @@
...
+import {Colors} from '../../theme/ColorsConstant'
+
+export const ExpandableWithSelection: React.FC<
+  ExpandableWithSelectionInterface
+> = ({
+  icon = '',
+  iconInfo = '',
+  expandableParagraph = '',
+  userRole = '',
+  successColor = Colors.backgroundColor,
```
Comment (ShubhamBade) [inline src/components/expandable/ExpandableWithSelection.tsx]: change color names - with proper name and update in interface as well

File: src/components/expandable/ExpandableWithSelection.tsx [hunk 16]
```diff
@@ -0,0 +1,173 @@
...
+  backgroundColour = Colors.backgroundColor,
+  fontSize = 2,
+  iconSize = 9,
+  iconColor = Colors.iconColor,
+  expandableTextSize = 2,
+  textColor = Colors.textColor,
+  borderColor = Colors.borderColor,
+  expandableTextColor = Colors.textColor,
+  imageSource = { uri: 'https://picsum.photos/700' },
+  imgSize = 40,
```
Comment (ShubhamBade) [inline src/components/expandable/ExpandableWithSelection.tsx]: imageSize

**Review by ShubhamBade — APPROVED** [review]
changes viewed.

**Review by Amitrawat21 — APPROVED** [review]

File: src/components/expandable/ExpandableWithSelection.tsx (hunk 10 above)
Comment (tabassum2507) [reply src/components/expandable/ExpandableWithSelection.tsx]: resolved

File: src/interface/expandable/expandableWithSelectionInterface.ts (hunk 12 above)
Comment (tabassum2507) [reply src/interface/expandable/expandableWithSelectionInterface.ts]: resolved

File: src/components/expandable/ExpandableWithSelection.tsx (hunk 15 above)
Comment (tabassum2507) [reply src/components/expandable/ExpandableWithSelection.tsx]: done

File: src/components/expandable/ExpandableWithSelection.tsx (hunk 16 above)
Comment (tabassum2507) [reply src/components/expandable/ExpandableWithSelection.tsx]: done

**Review by rohan-arya-w — APPROVED** [review]

**Pull request merged**